src/views/forms/FormNodeEdit/FormNodeEditTs.ts

```diff
@@ -0,0 +1,103 @@
+//@ts-ignore
+import FormRow from '@/components/FormRow/FormRow.vue'
+//@ts-ignore
+import { ValidationObserver, ValidationProvider } from 'vee-validate'
+// @ts-ignore
+import ErrorTooltip from '@/components/ErrorTooltip/ErrorTooltip.vue'
+//@ts-ignore
+import NetworkDelay from '@/components/NetworkDelay/NetworkDelay.vue'
+import { ValidationRuleset } from '@/core/validation/ValidationRuleset'
+import { URLHelpers } from '@/core/utils/URLHelpers'
+import { NotificationType } from '@/core/utils/NotificationType'
+import { mapGetters } from 'vuex'
+import { NodeModel } from '@/core/database/entities/NodeModel'
+import { NetworkType } from 'symbol-sdk'
+import { NetworkService } from '@/services/NetworkService'
+
+@Component({
+  components: {
+    ValidationObserver,
+    ValidationProvider,
+    ErrorTooltip,
+    FormRow,
+    NetworkDelay,
+  },
+  computed: {
+    ...mapGetters({
+      knowNodes: 'network/knowNodes',
+    }),
+  },
+})
+export class FormNodeEditTs extends Vue {
+  public formItems = {
+    todoType: 'add-node',
+    nodeUrl: '',
+    networkType: '',
+    networkHash: '',
+  }
+  public $refs!: {
+    observer: InstanceType<typeof ValidationObserver>
+  }
+  public validationRules = ValidationRuleset
+  public knowNodes: NodeModel[]
+  public nodeDelay: number = 0
+  public onSubmit() {
+    if (this.formItems.todoType === 'add-node') {
+      this.addPeer()
+      this.$emit('onCloseModal')
+    }
+  }
+  /**
+   * Add a new peer to storage
+   * @return {void}
+   */
+  public addPeer() {
+    // validate and parse input
+    const nodeUrl = URLHelpers.getNodeUrl(this.formItems.nodeUrl)
+
+    // return if node already exists in the database
+    if (this.knowNodes.find((node) => node.url === nodeUrl)) {
+      this.$store.dispatch('notification/ADD_ERROR', NotificationType.NODE_EXISTS_ERROR)
+      return
+    }
+
+    // read network type from node pre-saving
+    try {
+      // hide loading overlay
+      this.$store.dispatch('network/ADD_KNOWN_PEER', nodeUrl)
+      this.$store.dispatch('notification/ADD_SUCCESS', NotificationType.OPERATION_SUCCESS)
+      this.$store.dispatch('diagnostic/ADD_DEBUG', 'PeerSelector added peer: ' + nodeUrl)
+
+      // reset the form input
+      this.formItems.nodeUrl = ''
+      this.$refs.observer.reset()
+    } catch (e) {
+      // hide loading overlay
+      this.$store.dispatch('diagnostic/ADD_ERROR', 'PeerSelector unreachable host with URL: ' + nodeUrl)
+      this.$store.dispatch('notification/ADD_ERROR', NotificationType.ERROR_PEER_UNREACHABLE)
+    }
+  }
+  public async getInfoFromUrl() {
+    if (this.$refs.observer.fields.nodeUrl.invalid) {
+      this.formItems.networkHash = ''
+      this.formItems.networkType = ''
+      return
+    }
+    const startTime: number = new Date().getTime()
+    const networkService = new NetworkService()
+    const { networkModel } = await networkService.getNetworkModel(this.formItems.nodeUrl).toPromise()
```

Comment:
`getNetworkModel` will obfuscate the fact that a node is invalid, because if an error occurs, it will return default values.

So, here, if I put a valid URL that is not a valid endpoint, I would expect the UI to say: hey, your the wallet could not connect the URL you provided
but instead, it won't tell me anything, and fill values from a query of the default node

![image](https://user-images.githubusercontent.com/36499647/82632086-d6d38500-9c29-11ea-85de-c1afccee9eff.png)

Also, please test the case when peers with different `networkType` / `generationHash` are used

Reply:
I found this problem when I did it, But I'm not sure why this method does this  `getNetworkModel `
So I plan to submit an issue later to track this issue